src/main/java/com/couchbase/lite/store/SQLiteViewStore.java

```diff
@@ -250,48 +255,115 @@ public void updateIndex() throws CouchbaseLiteException {
+                    }
+
+                    int changes = 0;
+                    if (last == 0) {
+                        changes = store.getStorageEngine().delete(queryString("maps_#"), null, null);
+                    } else {
+                        store.optimizeSQLIndexes();
+                        String[] args = {Long.toString(last), Long.toString(last)};
+                        changes = store.getStorageEngine().delete(queryString("maps_#"),
+                                        "sequence IN (SELECT parent FROM revs " +
+                                        "WHERE sequence>? AND +parent>0 AND +parent<=?)", args);
+                    }
 
-            // First remove obsolete emitted results from the 'maps' table:
-            if (last < 0) {
-                String msg = String.format("last < 0 (%s)", last);
-                throw new CouchbaseLiteException(msg, new Status(Status.INTERNAL_SERVER_ERROR));
-            } else if (last < dbMaxSequence) {
-
-                minLastSequence = Math.min(minLastSequence, last);
-
-                if (last == 0) {
-                    // If the lastSequence has been reset to 0, make sure to remove any leftover rows:
-                    store.getStorageEngine().execSQL(queryString("DELETE FROM 'maps_#'"));
-                } else {
-                    store.optimizeSQLIndexes();
-                    // Delete all obsolete map results (ones from since-replaced revisions):
-                    String[] args = {Long.toString(last), Long.toString(last)};
-                    store.getStorageEngine().execSQL(
-                            queryString("DELETE FROM 'maps_#' WHERE sequence IN ("
-                                    + "SELECT parent FROM revs WHERE sequence>? "
-                                    + "AND +parent>0 AND +parent<=?)"), args);
+                    // Update #deleted rows:
+                    deletedCount += changes;
 
+                    // Only count these deletes as changes if this isn't a view reset to 0
+                    if (last != 0) {
+                        int newTotalRows = viewTotalRows.get(viewID) - changes;
+                        viewTotalRows.put(viewID, newTotalRows);
+                    }
                 }
             }
 
             if (minLastSequence == dbMaxSequence) {
-                // nothing to do (eg,  kCBLStatusNotModified)
                 Log.v(Log.TAG_VIEW, "minLastSequence (%s) == dbMaxSequence (%s), nothing to do",
                         minLastSequence, dbMaxSequence);
-                result.setCode(Status.NOT_MODIFIED);
-                return;
+                success = true;
+                return new Status(Status.NOT_MODIFIED);
             }
 
-            // This is the emit() block, which gets called from within the
-            // user-defined map() block
+            Log.v(Log.TAG_VIEW, "Updating indexes of (%s) from #%s to #%s ...",
+                    viewNames(views), minLastSequence, dbMaxSequence);
```

Comment:
http://docs.oracle.com/javase/7/docs/api/java/util/Formatter.html

```
's', 'S'	general	If the argument arg is null, then the result is "null". If arg implements Formattable, then arg.formatTo is invoked. Otherwise, the result is obtained by invoking arg.toString().
```

`long` -> `Long` -> `Long.toString()` could be happened here.